### PR TITLE
fix: issue with preserving new line characters in read-rule

### PR DESCRIPTION
**Pull request by armano2**
fixes: https://github.com/bradzacher/eslint-plugin-typescript/issues/249

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+        "node": 6
+      }
+    }]
   ],
-  env: {
-    test: {
-      ignore: null
+  "env": {
+    "test": {
+      "ignore": null
```
Comment (j-f1) [inline .babelrc]: Can you revert the changes to this file?

File: src/actions/__tests__/read-rule.test.js [hunk 2]
```diff
@@ -74,4 +74,45 @@ describe('readRule()', () => {
...
+  })
+
+  it('single line heading', () => {
+    const docs = `# Test`
+    const outputDocs = `# ${description} (${name})`
+
+    expect(runReadRule(undefined, docs).newDocs).toBe(outputDocs)
+  })
+
+  it('empty file', () => {
```
Comment (j-f1) [inline src/actions/__tests__/read-rule.test.js]: Can you rephrase these so they form complete sentences?

```suggestion
  it('handles an empty file properly', () => {
```

(If you need assistance, please don’t hesitate to ask for help)

File: src/actions/__tests__/read-rule.test.js [hunk 3]
```diff
@@ -74,4 +74,45 @@ describe('readRule()', () => {
       runReadRule(undefined, docs.replace('\n', '\r\n')).newDocs
     ).toMatchSnapshot()
   })
+
+  it('mixed windows and unix newlines', () => {
+    const docs = `# ${description} (${name})\nSome random test:\r\n\r\n`
```
Comment (j-f1) [inline src/actions/__tests__/read-rule.test.js]: Would it make sense to extract the `# ${description} (${name})` out to a `heading` variable so it doesn’t have to be repeated in each test?

File: src/actions/read-rule.js [hunk 4]
```diff
@@ -15,11 +15,13 @@ module.exports = ({ rule, docs, friendlyDocPath }, name) => {
   const fixable = rule.meta.fixable
   const { description, extraDescription, recommended } = rule.meta.docs
 
-  const nl = detectNewline(docs)
+  const heading = `# ${description} (${name})`
 
-  const newDocs = [`# ${description} (${name})`]
-    .concat(docs.split(nl).slice(1))
-    .join(nl)
+  const regTest = /^\s*#[^\r\n]*(\r?\n)?/
```
Comment (j-f1) [inline src/actions/read-rule.js]: Can you rename this `headingRe`?

Comment (j-f1) [reply src/actions/read-rule.js]: Or something else that makes more sense to you, as long as it clarifies that it’s matching the heading.

File: .babelrc (hunk 1 above)
Comment (armano2) [reply .babelrc]: .babelrc is json file but ok

**Comment by armano2** [comment]
@j-f1 suggestions applied

File: src/actions/__tests__/read-rule.test.js (hunk 2 above)
Comment (j-f1) [reply src/actions/__tests__/read-rule.test.js]: Looks like these changes didn’t get pushed.

Comment (armano2) [reply src/actions/__tests__/read-rule.test.js]: i see them in github https://github.com/j-f1/eslint-docs/pull/37/files#diff-1ef53d81915ab858a1a34aa1c5735e25R113

Comment (j-f1) [reply src/actions/__tests__/read-rule.test.js]: Ah, I see. Can you update the other test names too?

Comment (armano2) [reply src/actions/__tests__/read-rule.test.js]: i did update it a little, can you check and give me your remarks?

i hate naming stuff
